Fix errors detected by linters

Gusted opened the pull request:
- Running the [golangci-lint](https://github.com/golangci/golangci-lint/) over the code detected some errors. Unfortunately there are still some more errors(mostly error-checking + unused code in test files), so it cannot be enabled _yet_.
- Add go.sum
- Use [`strings.EqualFold`](https://pkg.go.dev/strings#EqualFold) instead of doing `strings.ToLower` twice, which is also a more correct-form to do case inequality checking.
-  Remove some dead-code that isn't exported and not used by the current code.
- Fixed a error that prevented the compiler to compile the code.
- Remove ineffective assignments(you already pass a pointer, so no need to assign the same pointer to the pointer).
- And run [gofumpt](https://github.com/mvdan/gofumpt) over the code to do some automatic code styling.

Commit:
Add go.sum

Commit:
Use strings.EqualFold

Commit:
Remove dead-code

Commit:
Fix compiling

Commit:
Remove ineffective assignments

Commit:
Automatic code-styling

mariusor commented:
Well, this is one year too late, and a little embarrassing.

@Gusted I apologise for missing this PR for so long. If you want to do another pass on top of most recent master, I'll take a look at it finally, and most likely, merge it.

I expected people to contact me on the mailing list [~mariusor/go-activitypub-dev@lists.sr.ht](mailto:~mariusor/go-activitypub-dev@lists.sr.ht) if they have patches or PRs, but I realize now, that for someone that doesn't look at the parent org, they might miss it.

mariusor committed:
Merge branch 'master' into linting

mariusor committed:
Revert some of the removed imports

mariusor committed:
Update decoding_json.go

mariusor committed:
No go.sum for the moment

We don't want to have go.sum in the repository while we haven't tagged at least one version.

mariusor committed:
Fixed whitespace

mariusor committed:
Fixed whitespace

mariusor reviewed (approved):
I made the changes required to merge this. I might revert some of the files, but overall looks good. 

Thank you @Gusted!
